content/en/blog/2025/expose-otel-collector-gateway-api/index.md

```diff
@@ -0,0 +1,601 @@
+
+In essence, Gateway API offers a more robust and standardized model for managing
+north-south traffic compared to the traditional Ingress API.
+
+## mTLS - A brief introduction
+
+Mutual TLS (mTLS) extends standard TLS by requiring _both_ the client and server
+to present and validate certificates for mutual authentication.
+
+Standard TLS (like HTTPS on websites) verifies the _server's_ identity to the
+_client_. Mutual TLS (mTLS) goes a step further:
+
+- The client verifies the server's identity (using the server's certificate).
+- The server _also_ verifies the client's identity (using the client's
+  certificate).
+
+Mutual TLS is important because it provides strong authentication, ensures
+end-to-end encryption, and aligns with Zero Trust security principles:
+
+- Ensures _only_ trusted clients (those possessing a valid certificate signed by
+  trusted CA) can connect to your exposed service.
+
+- All communication between the authenticated client and the server (such as a
+  Gateway) is encrypted.
+
+- It supports a Zero Trust model by requiring verification from both parties,
+  never assuming trust by default.
+
+## The Scenario
+
+1.  Deploy an OTel Collector inside Kubernetes configured with a simple OTLP
+    gRPC receiver.
+2.  Generate a self-signed Root CA, a server certificate (for the Gateway), and
+    a client certificate (for the external client).
+3.  Configure a Kubernetes `Gateway` resource to listen on a specific port,
+    terminate TLS, and require client certificates (mTLS).
+4.  Configure a `GRPCRoute` to route incoming gRPC traffic from the `Gateway` to
+    the internal OTel Collector service.
+5.  Configure an external client (another OTel Collector) to export data via
+    OTLP gRPC, using the client certificate and trusting the Root CA.
+
+![Scenario Diagram](ScenarioFlow.png)
+
+## Setup
+
+### Step 1: Install Gateway API CRD.
+
+By default, the Gateway APIs do not come installed in the Kubernetes Cluster. As
+of writing this blog, latest is
+[v1.2](https://gateway-api.sigs.k8s.io/implementations/v1.2/). Install the
+Gateway API CRDs if not present:
+
+```bash
+kubectl get crd gateways.gateway.networking.k8s.io &> /dev/null || \
+kubectl apply -f https://github.com/kubernetes-sigs/gateway-api/releases/download/v1.2.1/standard-install.yaml
+
+#To support GRPCRoute as of now. Would not be required once GRPCRoute CRD becomes GA.
+kubectl kustomize "github.com/kubernetes-sigs/gateway-api/config/crd/experimental?ref=v1.2.0" | kubectl apply -f -
+```
+
+### Step 2: Generate Self Signed Certificates
+
+In order to setup mTLS between the client and the server, we need a set of
+certificates. For this demo scenario, we'll be using self-signed certificate. We
+will be using the same CA for signing both clients and server in this demo.
+We'll use `openssl` to create our certificates. Please refer `openssl`
+documentation for configuration details.
```

Comment:
what would I do in a non demo scenario, can you outline that or provide some perspective, can something like cert-manager be used to automate/simplify this?